pkg/querier/queryrange/codec.go

```diff
@@ -82,8 +83,9 @@ func (r *LokiSeriesRequest) GetQuery() string {
 	return ""
 }
 
+// set default step value, this will be used in caching
 func (r *LokiSeriesRequest) GetStep() int64 {
-	return 0
+	return int64(time.Duration(int(math.Max(math.Floor(r.EndTs.Sub(r.StartTs).Seconds()/250), 1))) * time.Millisecond)
```

Comment:
This is required for `cache`, returning just `0` would make it fail here https://github.com/cortexproject/cortex/blob/1c764096b1ea0956005da64c17271ce58b1c9ce4/pkg/querier/queryrange/results_cache.go#L424 with `division by zero`

I hope this default is fine

Reply:
Hrm, I don't really love the idea of setting this here, but it should be fine.

Reply:
Can you do 1 ? instead